src/main/java/ui/MenuControl.java

```diff
@@ -89,47 +86,69 @@ private Menu createFileMenu() {
     }
 
     /**
-     * Logs the user action and creates a new board
+     * Checks whether the current open board has unsaved changes
+     * @return {@code true} if current open board is dirty, {@code false} otherwise.
      */
-    private void onBoardNew() {
-        logger.info("Menu: Boards > New");
-        newBoard();
+    public final boolean isCurrentOpenBoardDirty() {
+        if (!prefs.getLastOpenBoardPanelInfos().isPresent()) {
+            return true;
+        }
+        return !panels.getCurrentPanelInfos().equals(prefs.getLastOpenBoardPanelInfos().get());
     }
 
     /**
-     * Creates an empty board. User will be prompted to
-     * confirm the action if there are unclosed panels
+     * Prompts a {@code ConfirmChangesDialog} to ask the user whether or not to save the unsaved changes
+     * in the current board.
```

Comment:
If you want to refer to the class (need to be changed though), use `@link`. Also, this method does not need to know if the current board has unsaved changes. `Prompts ... whether or not to save the current board` should be sufficient.